netCDF4_utils.py

```diff
@@ -244,36 +244,37 @@ def _StartCountStride(elem, shape, dimensions=None, grp=None, datashape=None,\
             hasEllipsis = True
         # Replace boolean array with slice object if possible.
         elif ea.dtype.kind == 'b':
-            el = e.tolist()
-            if any(el):
-                start = el.index(True)
-                el.reverse()
-                stop = len(el)-el.index(True)
-                step = False
-                if e[start:stop].all():
-                    step = 1
-                else:
-                    n1 = start+1
-                    ee = e[n1]
-                    estart = e[start]
-                    while ee != estart:
-                        n1 = n1 + 1
-                        ee = e[n1]
-                    step = n1-start
-                    # check to make sure e[start:stop:step] are all True,
-                    # and other elements in e[start:stop] are all False.
-                    ii = range(start,stop,step)
-                    for i in range(start,stop):
-                        if i not in ii:
-                            if e[i]: step = False
-                        else:
-                            if not e[i]: step = False
-                if step: # it step False, can't convert to slice.
-                    newElem.append(slice(start,stop,step))
-                else:
-                    newElem.append(e)
-            else:
-                newElem.append(slice(0,0))
+           el = e.tolist()
```

Comment:
This is going to be a slow step regardless -- we really don't want to convert to lists if possible to avoid.